src/holofuel/pages/TransactionHistory/TransactionHistory.js

```diff
@@ -101,19 +109,24 @@ export default function TransactionsHistory ({ history: { push } }) {
     !loadingPendingTransactions &&
     !loadingCompletedTransactions
 
-  const partitionedTransactions = ([{
+  let partitionedTransactions = ([{
     label: 'Pending',
     transactions: filteredPendingTransactions,
     loading: loadingPendingTransactions
-  }]
-    .concat([{
-      // this partition only displays if completed transactions is loading
-      label: 'Completed',
-      transactions: [],
-      loading: loadingCompletedTransactions
-    }])
-    .concat(partitionByDate(filteredCompletedTransactions)))
-    .filter(({ transactions, loading }) => !isEmpty(transactions) || loading)
+  }])
+
+  let firstCompletedLabel, firstCompletedTransaction
+  if (partitionByDate(filteredCompletedTransactions)[0]) {
+    firstCompletedLabel = partitionByDate(filteredCompletedTransactions)[0].label
+    firstCompletedTransaction = partitionByDate(filteredCompletedTransactions)[0].transactions
+  }
+  const completedPartitionedTransactions = [{
+    label: firstCompletedLabel || 'Completed',
+    transactions: firstCompletedTransaction || [],
+    loading: loadingCompletedTransactions
+  }].concat(partitionByDate(filteredCompletedTransactions).slice(1))
+
+  partitionedTransactions = partitionedTransactions.concat(completedPartitionedTransactions).filter(({ transactions, loading }) => !isEmpty(transactions) || loading)
```

Comment:
this line could be
```
const partitionedTransactions = ([{
  label: 'Pending',
  transactions: filteredPendingTransactions,
  loading: loadingPendingTransactions
}])
  .concat(completedPartitionedTransactions).filter(({ transactions, loading }) => !isEmpty(transactions) || loading)
```

and remove the `let partitionedTransactions ...` above

Reply:
Good clean up remark.  Updating now. :+1: :